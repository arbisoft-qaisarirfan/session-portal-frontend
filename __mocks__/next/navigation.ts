export const useRouter = jest.fn(() => ({
  push: jest.fn(),
}));
export const useParams = jest.fn();
export const usePathname = jest.fn();
