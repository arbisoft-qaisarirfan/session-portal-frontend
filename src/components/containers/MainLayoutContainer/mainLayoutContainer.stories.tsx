import Box from "@mui/material/Box";
import Typography from "@mui/material/Typography";
import type { Meta, StoryObj } from "@storybook/react";

import MainLayoutContainer from "./mainLayoutContainer";

const meta: Meta<typeof MainLayoutContainer> = {
  title: "Layouts/MainLayoutContainer",
  component: MainLayoutContainer,
  tags: ["autodocs"],
  argTypes: {
    children: {
      control: "text",
      description: "The main content of the layout",
    },
    rightSidebar: {
      control: "object",
      description: "Optional right sidebar content",
    },
  },
};

export default meta;

type Story = StoryObj<typeof MainLayoutContainer>;

export const Default: Story = {
  args: {
    children: (
      <Box>
        <Typography variant="h4">Main Content</Typography>
        <Typography>This is the main content area. You can add any content here, such as text, images, or components.</Typography>
      </Box>
    ),
  },
};

export const WithRightSidebar: Story = {
  args: {
    ...Default.args,
    rightSidebar: (
      <Box>
        <Typography variant="h6">Right Sidebar</Typography>
        <Typography>This is the right sidebar content. It can include additional information, links, or widgets.</Typography>
      </Box>
    ),
  },
};

export const NoSidebar: Story = {
  args: {
    ...Default.args,
    rightSidebar: null,
  },
};
