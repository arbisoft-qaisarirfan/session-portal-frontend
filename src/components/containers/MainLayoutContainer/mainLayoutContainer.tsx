"use client";

import React, { isValidElement, ReactNode, useState } from "react";

import Box from "@mui/material/Box";

import Navbar from "@/components/Navbar";
import Sidebar from "@/components/Sidebar/sidebar";
import useAuth from "@/hooks/useAuth";

import { MainContainer, LeftSidebar, RightSidebar, ContentContainer, StyledDrawer } from "./styled";

type TMainLayoutContainer = {
  children: React.ReactNode;
  isLeftSidebarVisible?: boolean;
  rightSidebar?: ReactNode;
  shouldShowDrawer?: boolean;
};

const MainLayoutContainer = ({
  children,
  rightSidebar,
  isLeftSidebarVisible = true,
  shouldShowDrawer = false,
}: TMainLayoutContainer) => {
  useAuth();
  const [open, setOpen] = useState(false);

  const toggleDrawer = (newOpen: boolean) => () => {
    setOpen(newOpen);
  };

  return (
    <>
      <Navbar onDrawerToggle={toggleDrawer(!open)} shouldShowDrawer={shouldShowDrawer} />
      <MainContainer maxWidth="xl">
        {shouldShowDrawer && (
          <StyledDrawer open={open} onClose={toggleDrawer(false)} data-testid="drawer">
            <Box sx={{ width: 250, p: 1 }} role="presentation" onClick={toggleDrawer(false)}>
              <Sidebar />
            </Box>
          </StyledDrawer>
        )}
        {isLeftSidebarVisible && (
          <LeftSidebar data-testid="left-sidebar">
            <Sidebar />
          </LeftSidebar>
        )}
        <ContentContainer container>
          <Box width="100%">{children}</Box>
        </ContentContainer>
        {isValidElement(rightSidebar) && (
          <RightSidebar sx={{ display: { xs: "none", md: "block" } }} data-testid="right-sidebar">
            {rightSidebar}
          </RightSidebar>
        )}
      </MainContainer>
    </>
  );
};

export default MainLayoutContainer;
