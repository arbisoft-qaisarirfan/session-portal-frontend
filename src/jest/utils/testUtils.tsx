import React, { ReactElement } from "react";

import "@testing-library/jest-dom";
import { render, RenderOptions } from "@testing-library/react";

import ThemeProvider from "@/components/theme/theme-provider";
import { Providers } from "@/redux/store/provider";

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, "wrapper">) =>
  render(ui, {
    wrapper: ({ children }) => (
      <Providers>
        <ThemeProvider>{children}</ThemeProvider>
      </Providers>
    ),
    ...options,
  });

export * from "@testing-library/react";

export { customRender };
