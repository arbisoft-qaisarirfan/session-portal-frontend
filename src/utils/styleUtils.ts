export const shouldForwardProp = (propName: string) => !propName.startsWith("$");
