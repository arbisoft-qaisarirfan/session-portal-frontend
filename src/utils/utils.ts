export function secondsToTime(seconds: number): string {
  const hours = Math.floor(seconds / 3600);
  const minutes = Math.floor((seconds % 3600) / 60);

  const formatNumber = (num: number): string =>
    num.toLocaleString(undefined, {
      useGrouping: false,
      minimumIntegerDigits: 2,
    });

  return `${formatNumber(hours)}:${formatNumber(minutes)}`;
}
