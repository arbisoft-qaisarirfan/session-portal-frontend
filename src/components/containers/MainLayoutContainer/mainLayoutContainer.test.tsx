import { useSearchParams } from "next/navigation";

import { customRender as render, screen, waitFor } from "@/jest/utils/testUtils";

import MainLayoutContainer from "./mainLayoutContainer";

jest.mock("next/navigation", () => ({
  ...jest.requireActual("next/navigation"),
  useSearchParams: jest.fn(),
  usePathname: jest.fn().mockReturnValue("/some-path"),
}));

const mockNavigateTo = jest.fn();
jest.mock("@/hooks/useNavigation", () => ({
  __esModule: true,
  default: () => ({
    navigateTo: mockNavigateTo,
    getPageUrl: jest.fn((key) => key),
  }),
}));

jest.mock("@/hooks/useSidebar", () => ({
  __esModule: true,
  default: jest.fn(() => ({
    sidebarItems: [
      { id: 1, name: "Home" },
      { id: 2, name: "Settings" },
      { id: 3, name: "Profile" },
      { id: 4, name: "Logout" },
    ],
  })),
}));

describe("MainLayoutContainer", () => {
  beforeEach(() => {
    (useSearchParams as jest.Mock).mockReturnValue({
      get: jest.fn().mockReturnValue("test search"),
    });

    jest.clearAllMocks();
  });

  test("should renders Navbar component", () => {
    render(
      <MainLayoutContainer>
        <div>Test Content</div>
      </MainLayoutContainer>
    );

    expect(screen.getByTestId("navbar")).toBeInTheDocument();
  });

  test("should RightSidebar should be hidden on small screens (xs)", () => {
    global.innerWidth = 500;
    global.dispatchEvent(new Event("resize"));

    render(
      <MainLayoutContainer rightSidebar={<div>Right Content</div>}>
        <div>Test Content</div>
      </MainLayoutContainer>
    );

    expect(globalThis.innerWidth).toBe(500);

    waitFor(() => {
      expect(screen.queryByTestId("right-sidebar")).toHaveStyle("display:none");
    });
  });

  test("should RightSidebar should be visible on medium and larger screens (md+)", () => {
    global.innerWidth = 1024;
    window.dispatchEvent(new Event("resize"));

    render(
      <MainLayoutContainer rightSidebar={<div>Right Content</div>}>
        <div>Test Content</div>
      </MainLayoutContainer>
    );

    waitFor(() => {
      expect(screen.queryByTestId("right-sidebar")).toHaveStyle("display:block");
    });
  });

  test("should renders Sidebar component inside LeftSidebar", () => {
    render(
      <MainLayoutContainer>
        <div>Test Content</div>
      </MainLayoutContainer>
    );

    expect(screen.getByTestId("left-sidebar")).toBeInTheDocument();
  });

  test("should renders children inside ContentContainer", () => {
    render(
      <MainLayoutContainer>
        <div data-testid="content">Test Content</div>
      </MainLayoutContainer>
    );

    expect(screen.getByTestId("content")).toBeInTheDocument();
  });

  test("should does not render RightSidebar if rightSidebar prop is undefined", () => {
    render(
      <MainLayoutContainer>
        <div>Test Content</div>
      </MainLayoutContainer>
    );

    expect(screen.queryByTestId("right-sidebar")).not.toBeInTheDocument();
  });

  test("should renders RightSidebar when rightSidebar prop is provided", () => {
    render(
      <MainLayoutContainer rightSidebar={<div>Right Content</div>}>
        <div>Test Content</div>
      </MainLayoutContainer>
    );

    expect(screen.getByTestId("right-sidebar")).toBeInTheDocument();
  });

  test("should renders multiple children correctly", () => {
    render(
      <MainLayoutContainer>
        <div data-testid="child-1">Child 1</div>
        <div data-testid="child-2">Child 2</div>
      </MainLayoutContainer>
    );

    expect(screen.getByTestId("child-1")).toBeInTheDocument();
    expect(screen.getByTestId("child-2")).toBeInTheDocument();
  });

  test("should matches snapshot for UI consistency", () => {
    const { asFragment } = render(
      <MainLayoutContainer rightSidebar={<div>Right Content</div>}>
        <div>Test Content</div>
      </MainLayoutContainer>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
