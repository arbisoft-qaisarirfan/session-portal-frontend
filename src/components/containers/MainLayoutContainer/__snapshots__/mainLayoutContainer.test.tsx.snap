// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayoutContainer should matches snapshot for UI consistency 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-1mb2kr5-MuiPaper-root-MuiAppBar-root"
    data-testid="navbar"
    style="--Paper-shadow: var(--mui-shadows-4); --Paper-overlay: var(--mui-overlays-4);"
  >
    <div
      class="MuiContainer-root css-12f197v-MuiContainer-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular css-1u9o2lj-MuiToolbar-root"
      >
        <div
          class="css-co8q06-Logo-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-4uxqju-MuiSvgIcon-root"
            data-testid="YouTubeIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 15l5.19-3L10 9v6m11.56-7.83c.13.47.22 1.1.28 1.9.07.8.1 1.49.1 2.09L22 12c0 2.19-.16 3.8-.44 4.83-.25.9-.83 1.48-1.73 1.73-.47.13-1.33.22-2.65.28-1.3.07-2.49.1-3.59.1L12 19c-4.19 0-6.8-.16-7.83-.44-.9-.25-1.48-.83-1.73-1.73-.13-.47-.22-1.1-.28-1.9-.07-.8-.1-1.49-.1-2.09L2 12c0-2.19.16-3.8.44-4.83.25-.9.83-1.48 1.73-1.73.47-.13 1.33-.22 2.65-.28 1.3-.07 2.49-.1 3.59-.1L12 5c4.19 0 6.8.16 7.83.44.9.25 1.48.83 1.73 1.73z"
            />
          </svg>
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1c6flzx-MuiTypography-root"
          >
            Arbisoft Sessions Portal
          </h6>
        </div>
        <div
          class="MuiBox-root css-1rqqp05"
        >
          <form
            class="css-1yzrppa-Search-root"
          >
            <div
              class="MuiInputBase-root MuiInputBase-colorPrimary css-16stwic-MuiInputBase-root-StyledInputBase-root"
            >
              <input
                aria-label="search"
                class="MuiInputBase-input css-1943rd7-MuiInputBase-input"
                data-testid="search-query"
                placeholder="Search..."
                type="text"
                value=""
              />
            </div>
            <button
              class="css-ejiypo-SearchIconWrapper-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-4uxqju-MuiSvgIcon-root"
                data-testid="SearchIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14"
                />
              </svg>
            </button>
          </form>
        </div>
        <div
          class="MuiBox-root css-1v32olg"
        >
          <button
            aria-label="Open settings"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1j2eanh-MuiButtonBase-root-MuiIconButton-root"
            data-mui-internal-clone-element="true"
            data-testid="avatar-btn"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiAvatar-root MuiAvatar-rounded MuiAvatar-colorDefault css-5ta4e5-MuiAvatar-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-pamvpt-MuiSvgIcon-root-MuiAvatar-fallback"
                data-testid="PersonIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                />
              </svg>
            </div>
          </button>
        </div>
      </div>
    </div>
  </header>
  <div
    class="MuiContainer-root MuiContainer-maxWidthXl css-18is7ce-MuiContainer-root-MainContainer-root"
  >
    <div
      class="MuiBox-root css-1i3wj7r-LeftSidebar-root"
      data-testid="left-sidebar"
    >
      <div
        class="MuiBox-root css-cd9kc0-SidebarContainer-root"
        data-testid="sidebar-container"
      >
        <div
          class="MuiStack-root css-ujfjh4-MuiStack-root-MenuStack-root"
        >
          <div
            class="MuiStack-root css-okam97-MuiStack-root-MenuItem-root"
          >
            <img
              alt="Home"
              data-nimg="1"
              decoding="async"
              height="12"
              loading="lazy"
              src="/assets/images/sidebar-item-icon.svg"
              style="color: transparent;"
              width="18"
            />
            <p
              class="MuiTypography-root MuiTypography-bodyMedium css-rudw3m-MuiTypography-root-Text-root"
              data-testid="sidebar-item-Home"
            >
              Home
            </p>
          </div>
          <div
            class="MuiStack-root css-okam97-MuiStack-root-MenuItem-root"
          >
            <img
              alt="Settings"
              data-nimg="1"
              decoding="async"
              height="12"
              loading="lazy"
              src="/assets/images/sidebar-item-icon.svg"
              style="color: transparent;"
              width="18"
            />
            <p
              class="MuiTypography-root MuiTypography-bodyMedium css-rudw3m-MuiTypography-root-Text-root"
              data-testid="sidebar-item-Settings"
            >
              Settings
            </p>
          </div>
          <div
            class="MuiStack-root css-okam97-MuiStack-root-MenuItem-root"
          >
            <img
              alt="Profile"
              data-nimg="1"
              decoding="async"
              height="12"
              loading="lazy"
              src="/assets/images/sidebar-item-icon.svg"
              style="color: transparent;"
              width="18"
            />
            <p
              class="MuiTypography-root MuiTypography-bodyMedium css-rudw3m-MuiTypography-root-Text-root"
              data-testid="sidebar-item-Profile"
            >
              Profile
            </p>
          </div>
          <div
            class="MuiStack-root css-okam97-MuiStack-root-MenuItem-root"
          >
            <img
              alt="Logout"
              data-nimg="1"
              decoding="async"
              height="12"
              loading="lazy"
              src="/assets/images/sidebar-item-icon.svg"
              style="color: transparent;"
              width="18"
            />
            <p
              class="MuiTypography-root MuiTypography-bodyMedium css-rudw3m-MuiTypography-root-Text-root"
              data-testid="sidebar-item-Logout"
            >
              Logout
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row css-1zhuio-MuiGrid2-root-ContentContainer-root"
    >
      <div>
        Test Content
      </div>
    </div>
    <div
      class="MuiBox-root css-1gxza0d-RightSidebar-root"
      data-testid="right-sidebar"
    >
      <div>
        Right Content
      </div>
    </div>
  </div>
</DocumentFragment>
`;
